credential.js

```diff
@@ -146,22 +146,33 @@ var crypto = require('crypto'),
    *
    * Compare two strings, x and y with a constant-time
    * algorithm to prevent attacks based on timing statistics.
-   * 
+   *
+   * This really ought to be in C; see:
+   *     https://github.com/joyent/node/issues/8560
+   *     http://stackoverflow.com/questions/18476402
+   *
+   * See also:
+   *     http://jsperf.com/constant-time-string-comparison
+   *     
    * @param  {String} x
    * @param  {String} y
    * @return {Boolean}
    */
-  constantEquals = function constantEquals(x, y) {
-    var result = true,
-      length = (x.length > y.length) ? x.length : y.length,
-      i;
-
-    for (i=0; i<length; i++) {
-      if (x.charCodeAt(i) !== y.charCodeAt(i)) {
-        result = false;
+  constantEquals = function constantEquals(a, b) {
+    // Using with{} nixes some V8 optimizations that would otherwise undermine
+    // our intentions here.
+    with({}) {
```

Comment:
Can you give me a detailed account of your reason for using `with` like this here? I believe that an optimizing compiler may see that `this` isn't used at all in this code and completely skip allocating the `with` object in any case, and if so, what is the purpose of `with`?